src/gui/transferlistsortmodel.h

```diff
@@ -56,7 +56,11 @@ class TransferListSortModel final : public QSortFilterProxyModel
     bool lessThan(const QModelIndex &left, const QModelIndex &right) const override;
     bool filterAcceptsRow(int sourceRow, const QModelIndex &sourceParent) const override;
     bool matchFilter(int sourceRow, const QModelIndex &sourceParent) const;
-    bool lessThan_impl(const QModelIndex &left, const QModelIndex &right) const;
+
+    // > 0 : left > right
+    // < 0 : left < right
+    // == 0 : failed or left == right
```

Comment:
What does `failed` mean here?
There used to be some logic here when comparing some "special" values (e.g. invalid dates), but you just declared it incomparable. It doesn't look like you're seriously considering it. Can you justify these changes (at least by giving some examples that demonstrate that the new logic is at least as good as the existing one)?

Reply:
`failed` means that any of the values are invalid or the values are not comparable. It flags the user of the function to fallback to some other criteria for comparison. by this logic all items with invalid values of this criteria will be grouped together when sorting and then they will be automatically sub-sorted with different criteria using the fallbacks introduced in lessThan functions.

Reply:
> There used to be some logic here when comparing some "special" values (e.g. invalid dates), but you just declared it incomparable

There were so many different criteria used here that it would be very difficult for the user to actually determine which criteria was used. IMO not treating these "special" values prefrentially, we make sorting more deterministically and simplify the code at the same time.

Reply:
hmm, the way I am implementing the "failed" state will break sorting itself.

Reply:
You should fallback if and only if previously compared values are really equal.